File: client/src/app/models/shopStats.ts
export interface ShopStats {
    annual: StatsElement;
    monthly: StatsElement;
    weekly: StatsElement;
    daily: StatsElement;
}


export interface StatsElementData {
    date: string;
    incoming: number;
    outgoing: number;

}

export interface StatsElement {
    title: string;
    elementData: StatsElementData[];
    totalIncoming: number;
    totalOutgoing: number;
    totalPurchases: number,
    totalPurchasesPaid: number,
    totalSales: number,
    totalSalesPaid: number,
    totalLoss: number,
}

File: API/Models/ShopChartsData/ShopStats.cs
namespace API.Models.ShopChartsData;

public class ShopStats
{
    public StatsElement Annual { get; set; } = new() {Title = "Annual"};
    public StatsElement Monthly { get; set; } = new() {Title = "Monthly"};
    public StatsElement Weekly { get; set; } = new() {Title = "Weekly"};
    public StatsElement Daily { get; set; } = new() {Title = "Daily"};
}